server/types/uuid.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"bytes"
+	"reflect"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+	"github.com/dolthub/vitess/go/sqltypes"
+	"github.com/dolthub/vitess/go/vt/proto/query"
+
+	"github.com/dolthub/doltgresql/postgres/parser/uuid"
+)
+
+// Uuid is the UUID type.
+var Uuid = UuidType{}
+
+// UuidType is the extended type implementation of the PostgreSQL UUID.
+type UuidType struct{}
+
+var _ types.ExtendedType = UuidType{}
+
+// CollationCoercibility implements the types.ExtendedType interface.
+func (b UuidType) CollationCoercibility(ctx *sql.Context) (collation sql.CollationID, coercibility byte) {
+	return sql.Collation_binary, 5
+}
+
+// Compare implements the types.ExtendedType interface.
+func (b UuidType) Compare(v1 any, v2 any) (int, error) {
+	if v1 == nil && v2 == nil {
+		return 0, nil
+	} else if v1 != nil && v2 == nil {
+		return 1, nil
+	} else if v1 == nil && v2 != nil {
+		return -1, nil
+	}
+
+	ac, _, err := b.Convert(v1)
+	if err != nil {
+		return 0, err
+	}
+	bc, _, err := b.Convert(v2)
+	if err != nil {
+		return 0, err
+	}
+
+	ab := ac.(uuid.UUID)
+	bb := bc.(uuid.UUID)
+	return bytes.Compare(ab.GetBytesMut(), bb.GetBytesMut()), nil
+}
+
+// Convert implements the types.ExtendedType interface.
+func (b UuidType) Convert(val any) (any, sql.ConvertInRange, error) {
+	if val == nil {
+		return nil, sql.InRange, nil
+	}
+
+	switch val := val.(type) {
+	case string:
+		uuidVal, err := uuid.FromString(val)
```

Comment:
Same question. Why are we using the parser type instead of the golang library type?

Reply:
Refer to previous comment